### PR TITLE
Implement dot_rot_grad_Ynm()

### PR DESCRIPTION
**Pull request by chakpongchung**
This implements  the formula described in the function dot_rot_grad_Ynm()

### CODE REVIEW

**Comment by chakpongchung** [comment]
Hi @certik  can you review it again?

**Comment by certik** [comment]
@chakpongchung I left a comment. After you fix it and if tests pass, we can merge this.

**Comment by chakpongchung** [comment]
@certik  arguments test fixed

File: sympy/core/tests/test_args.py [hunk 1]
```diff
@@ -3673,3 +3673,7 @@ def test_sympy__vector__scalar__BaseScalar():
     from sympy.vector.coordsysrect import CoordSysCartesian
     C = CoordSysCartesian('C')
     assert _test_args(BaseScalar('Cx', 0, C, ' ', ' '))
+
+def test_sympy__physics__wigner__Wigner3j():
+    from sympy.physics.wigner import Wigner3j
+    assert _test_args(Wigner3j(0,0,0,0,0,0))
```
Comment (certik) [inline sympy/core/tests/test_args.py]: Can you add space after each comma (`,`)?

File: sympy/physics/tests/test_clebsch_gordan.py [hunk 2]
```diff
@@ -127,3 +128,24 @@ def test_racah():
...
+    theta, phi = symbols("theta phi")
+    assert dot_rot_grad_Ynm(1, 1, 1, 1, 1, 0) !=  \
+        sqrt(30)*Ynm(2, 2, 1, 0)/(10*sqrt(pi))
+    assert dot_rot_grad_Ynm(1, 1, 1, 1, 1, 0).doit() ==  \
+        sqrt(30)*Ynm(2, 2, 1, 0)/(10*sqrt(pi))
+    assert dot_rot_grad_Ynm(1, 5, 1, 1, 1, 2) !=  \
+        0
+    assert dot_rot_grad_Ynm(1, 5, 1, 1, 1, 2).doit() ==  \
+        0
+    assert dot_rot_grad_Ynm(3,3,3,3,theta,phi).doit() ==  \
```
Comment (certik) [inline sympy/physics/tests/test_clebsch_gordan.py]: Space after commas. E.g. make it `3, 3, 3, 3, theta, phi`.

File: sympy/physics/tests/test_clebsch_gordan.py [hunk 3]
```diff
@@ -127,3 +128,24 @@ def test_racah():
...
+        15*sqrt(3003)*Ynm(6, 6, theta, phi)/(143*sqrt(pi))
+    assert dot_rot_grad_Ynm(3,3,1,1,theta,phi).doit() ==  \
```
Comment (certik) [inline sympy/physics/tests/test_clebsch_gordan.py]: The same here.

File: sympy/physics/tests/test_clebsch_gordan.py [hunk 4]
```diff
@@ -127,3 +128,24 @@ def test_racah():
...
+        sqrt(3)*Ynm(4, 4, theta, phi)/sqrt(pi)
+    assert dot_rot_grad_Ynm(3,2,2,0,theta,phi).doit() ==  \
```
Comment (certik) [inline sympy/physics/tests/test_clebsch_gordan.py]: The same here.

File: sympy/physics/tests/test_clebsch_gordan.py [hunk 5]
```diff
@@ -127,3 +128,24 @@ def test_racah():
...
+        3*sqrt(55)*Ynm(5, 2, theta, phi)/(11*sqrt(pi))
+    assert dot_rot_grad_Ynm(3,2,3,2,theta,phi).doit() ==  \
```
Comment (certik) [inline sympy/physics/tests/test_clebsch_gordan.py]: The same here.

**Comment by certik** [comment]
Thanks. I left a few comments for some minor things to fix. Otherwise I don't have any more suggestions and once tests pass, we can merge.

**Comment by chakpongchung** [comment]
coding style fixed, i will follow the pep8 later on as suggested by Harsh.
Thanks!

**Comment by certik** [comment]
Perfect!

+1 to merge if tests pass.

**Comment by certik** [comment]
Tests pass, merging.

**Comment by certik** [comment]
@chakpongchung if you have time, would you mind sending a another PR where you would add the equations that `dot_rot_grad_Ynm()` solves? In particular, add the definition of `alpha` and `R`. So that it is clear what the left hand side is, and what the right hand side is.

**Comment by chakpongchung** [comment]
Sure.
i will try to add more details to document it.

**Comment by certik** [comment]
@chakpongchung thank you. Let's get it done before you forget the details. Please ping me if you get stuck.

**Comment by chakpongchung** [comment]
Thanks.But the work is still going on. Will let you know when appropriate

**Comment by certik** [comment]
@chakpongchung just to clarify what I am asking for --- I am asking to add the equations for `R` and `alpha`, which you already have in the pdf that you showed me. It needs to be added into the docstring to make it complete.

**Pull request merged**